src/Phase.jl

```diff
@@ -79,15 +79,16 @@ function IdealDiluteSolution(species,reactions,solvent; name="",diffusionlimited
     rxns = [ElementaryReaction(index=i,reactants=rxn.reactants,reactantinds=rxn.reactantinds,products=rxn.products,
         productinds=rxn.productinds,kinetics=rxn.kinetics,radicalchange=rxn.radicalchange,reversible=rxn.reversible,pairs=rxn.pairs) for (i,rxn) in enumerate(rxns)]
     therm = getvecthermo(species)
-    M,Nrp = getstoichmatrix(species,rxns)
+    rxnarray = getreactionindices(species,rxns)
+    M,Nrp = getstoichmatrix(rxnarray,species)
     echangevec = getfield.(rxns,:electronchange)
     if all(echangevec .== 0)
         electronchange = nothing
     else 
         electronchange = convert(echangevec,Array{Float64,1})
     end
     reversibility = getfield.(rxns,:reversible)
-    rxnarray = getreactionindices(species,rxns)
+
     return IdealDiluteSolution(species=species,reactions=rxns,solvent=solvent,name=name,
         spcdict=Dict([sp.name=>sp.index for sp in species]),stoichmatrix=M,Nrp=Nrp,rxnarray=rxnarray,veckinetics=vectuple,
```

Comment:
Should this be `Dict([sp.name=>i for (i,sp) in enumerate(species)])` if we are avoiding to use the species' index to indicate the species's position in an array?

Reply:
fixed